runtime/lua/vim/iter.lua

```diff
@@ -112,6 +112,21 @@ local function sanitize(t)
   return t
 end
 
+-- this returns a table that contains the flattened v
```

Comment:
Please add a proper docstring to this function (including type annotations for parameters)